microbench/src/main/java/io/netty/microbenchmark/common/IsValidIpV6Benchmark.java

```diff
@@ -0,0 +1,306 @@
+                if (!isValidIp4Word(word.toString())) {
+                    return false;
+                }
+                if (numberOfColons != 6 && !doubleColon) {
+                    return false;
+                }
+                // a special case ::1:2:3:4:5:d.d.d.d allows 7 colons with an
+                // IPv4 ending, otherwise 7 :'s is bad
+                if (numberOfColons == 7 && ipAddress.charAt(startOffset) != ':' &&
+                    ipAddress.charAt(1 + startOffset) != ':') {
+                    return false;
+                }
+                word.delete(0, word.length());
+                break;
+
+            case ':':
+                // FIX "IP6 mechanism syntax #ip6-bad1"
+                // An IPV6 address cannot start with a single ":".
+                // Either it can start with "::" or with a number.
+                if (i == startOffset && (ipAddress.length() <= i || ipAddress.charAt(i + 1) != ':')) {
+                    return false;
+                }
+                // END FIX "IP6 mechanism syntax #ip6-bad1"
+                numberOfColons ++;
+                if (numberOfColons > 7) {
+                    return false;
+                }
+                if (numberOfPeriods > 0) {
+                    return false;
+                }
+                if (prevChar == ':') {
+                    if (doubleColon) {
+                        return false;
+                    }
+                    doubleColon = true;
+                }
+                word.delete(0, word.length());
+                break;
+
+            default:
+                if (word != null && word.length() > 3) {
+                    return false;
+                }
+                if (!isValidHexChar(c)) {
+                    return false;
+                }
+                word.append(c);
+            }
+        }
+
+        // Check if we have an IPv4 ending
+        if (numberOfPeriods > 0) {
+            // There is a test case with 7 colons and valid ipv4 this should resolve it
+            if (numberOfPeriods != 3 || !(isValidIp4Word(word.toString()) && numberOfColons < 7)) {
+                return false;
+            }
+        } else {
+            // If we're at then end and we haven't had 7 colons then there is a
+            // problem unless we encountered a doubleColon
+            if (numberOfColons != 7 && !doubleColon) {
+                return false;
+            }
+
+            // If we have an empty word at the end, it means we ended in either
+            // a : or a .
+            // If we did not end in :: then this is invalid
+            if (word.length() == 0 && ipAddress.charAt(length - 1 - startOffset) == ':' &&
+                ipAddress.charAt(length - 2 - startOffset) != ':') {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    // New
+
+    private static boolean isValidIpV4Word_new(CharSequence word, int start, int endExclusive) {
```

Comment:
can you follow this recommendation? `isValidIpV4WordNew` should be good enough. Lets do the same for other comments about method names.